numpy/lib/function_base.py

```diff
@@ -4445,19 +4492,22 @@ def _lerp(a, b, t, out=None):
 
 def _get_gamma_mask(shape, default_value, conditioned_value, where):
     out = np.full(shape, default_value)
-    out[where] = conditioned_value
+    np.copyto(out, conditioned_value, where=where, casting="unsafe")
```

Comment:
Normally `condition_value` is just a scalar, so this doesn't matter.  But in one of the branches, it is an array that matches the shape of `out`, so we have to use the `copyto` logic of copying both.  (To be honest, the casting is just for sports right now.)